mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -672,7 +673,7 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
         itemFactors = computeFactors(userFactors, userOutBlocks, itemInBlocks, rank, regParam,
           userLocalIndexEncoder, solver = solver)
         if (shouldCheckpoint(iter)) {
-          itemFactors.checkpoint()
+          ALS.checkpointAndCleanParents(itemFactors)
           itemFactors.count() // checkpoint item factors and cut lineage
           deletePreviousCheckpointFile()
```

Comment:
@mengxr I haven't looked at this code in quite a while - can you remind me why in the explicit case `deletePreviousCheckpointFile` and `previousCheckpointFile = itemFactors.getCheckpointFile` happens before `userFactors` is computed, while in the implicit case it happens after?